src/Zns.ts

```diff
@@ -246,4 +247,36 @@ export default class Zns extends NamingService {
     const record = await this.getContractField(contractAddress, field, [key]);
     return (record && record[key]) || null;
   }
+
+  private checkNetworkConfig(source: ZnsSource): void {
+    if (!source.network) {
+      throw new ConfigurationError(ConfigurationErrorCode.UnspecifiedNetwork, {
+        method: NamingServiceName.CNS,
+      });
+    }
+    if (!ZnsSupportedNetwork.guard(source.network)) {
+      this.checkCustomNetworkConfig(source);
+    }
+  }
+
+  private checkCustomNetworkConfig(source: ZnsSource): void {
+    if (!source.registryAddress) {
+      throw new ConfigurationError(
+        ConfigurationErrorCode.CustomNetworkConfigMissing,
+        {
+          method: NamingServiceName.CNS,
+          config: 'registryAddress',
+        },
+      );
+    }
+    if (!source['url'] && source['provider']) {
```

Comment:
Should it look like this?

```suggestion
    if (!source['url'] || !source['provider']) {
```